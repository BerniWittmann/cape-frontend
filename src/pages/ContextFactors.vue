<template>
  <v-layout>
    <el-row>
      <el-col :span="20" :offset="2">
        <div class="contextFactors">
          <h3>{{$t('context_factor.title')}}</h3>
          <el-tree :data="contextTree" ></el-tree>
        </div>
      </el-col>
    </el-row>
  </v-layout>
</template>

<script>
import DefaultLayout from '@/layouts/Default.vue'

export default {
  components: {
    VLayout: DefaultLayout
  },
  methods: {
    getChildren(contextFactor) {
      return this.allContextFactors.filter(cF => cF.parentID === contextFactor.id)
    },
    constructChildren(contextFactor) {
      let a = this.getChildren(contextFactor)
      let b = []
      for (let i = 0; i < a.length; i++) {
        b.push({ label: a[i].name, contextFactor: a[i], children: this.constructChildren(a[i]) })
      }
      return b
    }
  },
  computed: {
    allContextFactors() {
      return this.$store.state.contextFactor.contextFactors
    },
    root() {
      return this.allContextFactors.find(cF => typeof cF.parentID === 'undefined')
    },
    contextTree() {
      return [{
        contextFactor: this.root,
        children: this.constructChildren(this.root),
        label: this.root.name
      }]
    }
  }
}

</script>

<style scoped lang="scss">
.contextFactors {
  text-align: center;
}
</style>
