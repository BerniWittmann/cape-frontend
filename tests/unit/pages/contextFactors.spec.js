import { shallowMount } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import ContextFactors from '@/pages/ContextFactors.vue'

describe('Pages', () => {
  describe('ContextFactors.vue', () => {
    let store
    let router
    let message
    let cmp

    beforeEach(() => {
      router = {
        push: jest.fn()
      }
      message = {
        error: jest.fn()
      }
      store = {
        state: {
          contextFactor: {
            contextFactors: [{
              attributes: [],
              id: '5c3c83a5a0983a6a94272513',
              name: 'Pizza Donalds',
              __v: 0
            }, {
              attributes: [],
              id: '5c3c83a5a0983a6a94272514',
              name: 'Oven',
              parentID: '5c3c83a5a0983a6a94272513',
              __v: 0
            }, {
              attributes: [],
              id: '5c3c83a5a0983a6a94272515',
              name: 'Fryer',
              parentID: '5c3c83a5a0983a6a94272513',
              __v: 0
            }, {
              attributes: [],
              id: '5c3c83a5a0983a6a94272516',
              name: 'Telephone',
              parentID: '5c3c83a5a0983a6a94272513',
              __v: 0
            }, {
              attributes: [],
              id: '5c3c83a5a0983a6a94272517',
              name: 'Temperature Sensor',
              parentID: '5c3c83a5a0983a6a94272514',
              __v: 0
            }, {
              attributes: [],
              id: '5c3c83a5a0983a6a94272518',
              name: 'Serial Number',
              parentID: '5c3c83a5a0983a6a94272514',
              __v: 0
            }, {
              attributes: [],
              id: '5c3c83a5a0983a6a94272519',
              name: 'Ventilator',
              parentID: '5c3c83a5a0983a6a94272514',
              __v: 0
            }]
          }
        }
      }
      render()
    })

    function render() {
      cmp = shallowMount(ContextFactors, {
        i18n,
        mocks: {
          $store: store,
          $router: router,
          $message: message
        }
      })
    }

    it('renders', () => {
      expect(cmp.html()).toMatchSnapshot()
    })
  })
})
