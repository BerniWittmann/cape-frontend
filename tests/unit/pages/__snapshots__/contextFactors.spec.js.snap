// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages ContextFactors.vue renders 1`] = `
<vlayout-stub>
  <elrow-stub tag="div" justify="start" align="top">
    <elcol-stub span="20" tag="div" offset="2">
      <div class="contextFactors">
        <h3>context_factor.title</h3>
        <eltree-stub data="[object Object]" emptytext="暂无数据" renderafterexpand="true" expandonclicknode="true" autoexpandparent="true" props="[object Object]" indent="18"></eltree-stub>
      </div>
    </elcol-stub>
  </elrow-stub>
</vlayout-stub>
`;
